Error: kill ESRCH

## Is this a bug report or a feature request?

Bug report

If this is a bug report, please provide as much info as possible

- version `1.4.1`
- platform: `teamcity`

*expected behavior*

The server to run in CI environment.

*actual behavior*

The command crashes with the following info:

```
[04:09:56][Step 3/6] { Error: kill ESRCH
[04:09:56][Step 3/6]     at exports._errnoException (util.js:1018:11)
[04:09:56][Step 3/6]     at process.kill (internal/process.js:190:13)
[04:09:56][Step 3/6]     at children.forEach.child (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/start-server-and-test/src/index.js:32:21)
[04:09:56][Step 3/6]     at Array.forEach (native)
[04:09:56][Step 3/6]     at Promise.fromNode.then.children (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/start-server-and-test/src/index.js:31:20)
[04:09:56][Step 3/6]     at tryCatcher (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/util.js:16:23)
[04:09:56][Step 3/6]     at Promise._settlePromiseFromHandler (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/promise.js:512:31)
[04:09:56][Step 3/6]     at Promise._settlePromise (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/promise.js:569:18)
[04:09:56][Step 3/6]     at Promise._settlePromise0 (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/promise.js:614:10)
[04:09:56][Step 3/6]     at Promise._settlePromises (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/promise.js:693:18)
[04:09:56][Step 3/6]     at Promise._fulfill (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/promise.js:638:18)
[04:09:56][Step 3/6]     at /home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/bluebird/js/release/nodeback.js:42:21
[04:09:56][Step 3/6]     at /home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/ps-tree/index.js:86:7
[04:09:56][Step 3/6]     at Stream.a.end (/home/teamcity/buildAgent/work/7182f89e35c6b1c9/node_modules/event-stream/index.js:87:5)
[04:09:56][Step 3/6]     at Stream.onend (internal/streams/legacy.js:44:10)
[04:09:56][Step 3/6]     at emitNone (events.js:91:20) code: 'ESRCH', errno: 'ESRCH', syscall: 'kill' }
[04:09:56][Step 3/6] error An unexpected error occurred: "Command failed.
[04:09:56][Step 3/6] Exit code: 1
[04:09:56][Step 3/6] Command: sh
[04:09:56][Step 3/6] Arguments: -c start-server-and-test start http://localhost:3000 cypress:run:ci
[04:09:56][Step 3/6] Directory: /home/teamcity/buildAgent/work/7182f89e35c6b1c9
[04:09:56][Step 3/6] Output:
[04:09:56][Step 3/6] ".
```

Here are the relevant commands I'm running:

```
"watch-css": "npm run build-css && node-sass-chokidar src/css/scss/ -o src/css/dist/ --watch",
"start": "npm-run-all -p watch-css start-js",
"cypress:run:ci": "./node_modules/.bin/cypress run",
"cypress:ci": "start-server-and-test start http://localhost:3000 cypress:run:ci"
```

`cypress:ci` being the one called in teamcity.

When I run this command locally, it works fine.


Thank you!

I'm running in to the same issue but I'm using Docker (and `cypress/base:8` as the base image)

* version: `1.7.0`
* platform: `Debian running in a docker container`

Here are my scripts:

```
"e2e:ci": "start-server-and-test start:ci http-get://localhost:8080 e2e:headless",
"start:ci": "npm start > /dev/null",
"start": "concurrently --kill-others --raw \"npm:ui\" \"npm:api\" ",
"ui": "vue-cli-service serve",
"api": "nodemon dev-server.js",
"e2e:headless": "cypress run --reporter mocha-multi-reporters --reporter-options configFile=tests/e2e/reporters.json --config watchForFileChanges=false",
```

and here is the error output:

```
{ Error: kill ESRCH
    at Object._errnoException (util.js:992:11)
    at process.kill (internal/process.js:183:18)
    at children.forEach.child (/app/node_modules/start-server-and-test/src/index.js:38:21)
    at Array.forEach (<anonymous>)
    at Promise.fromNode.then.children (/app/node_modules/start-server-and-test/src/index.js:37:20)
    at tryCatcher (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (/app/node_modules/start-server-and-test/node_modules/bluebird/js/release/promise.js:638:18)
    at /app/node_modules/start-server-and-test/node_modules/bluebird/js/release/nodeback.js:42:21
    at /app/node_modules/ps-tree/index.js:86:7
    at Stream.a.end (/app/node_modules/event-stream/index.js:87:5)
    at Stream.onend (internal/streams/legacy.js:44:10)
    at emitNone (events.js:111:20) code: 'ESRCH', errno: 'ESRCH', syscall: 'kill' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vision-ui@1.1.0-dev e2e:ci: `start-server-and-test start:ci http-get://localhost:8080 e2e:headless`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the vision-ui@1.1.0-dev e2e:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```

--- a/src/index.js
+++ b/src/index.js
@@ -38,7 +38,11 @@ function startAndTest ({ start, url, test }) {
             try {
               process.kill(child.PID)
             } catch (e) {
-              // The process may already stop
+              if (e.code === 'ESRCH') {
+                console.log(`Child process ${child.PID} exited before trying to stop it`)
+              } else {
+                throw e
+              }
             }
           })
         })

--- a/src/index.js
+++ b/src/index.js
@@ -35,7 +35,11 @@ function startAndTest ({ start, url, test }) {
         .then(children => {
           debug('stopping child processes')
           children.forEach(child => {
-            process.kill(child.PID)
+            try {
+              process.kill(child.PID)
+            } catch (e) {
+              // The process may already stop
+            }
           })
         })
         .then(() => {

